src/poetry/core/version/markers.py

```diff
@@ -294,9 +294,9 @@ def exclude(self, marker_name: str) -> BaseMarker:
 
         return self
 
-    def only(self, *marker_names: str) -> Union["SingleMarker", EmptyMarker]:
+    def only(self, *marker_names: str) -> Union["SingleMarker", AnyMarker]:
```

Comment:
The commit mentions `get_python_constraint_from_marker()` and `SingleMarker.only()`, is it missing a change?

Reply:
`get_python_constraint_from_marker()` is fixed indirectly by fixing `SingleMarker.only()` which is used in `get_python_constraint_from_marker()`:
https://github.com/python-poetry/poetry-core/blob/e8031bbc2d573f94a99b0d0ab4d25f8efcef83f8/src/poetry/core/packages/utils/utils.py#L303

Without the change both tests (for `SingleMarker.only()` and `get_python_constraint_from_marker()`) will fail.